Lib/string.py

```diff
@@ -64,14 +71,24 @@ class Template:
     # See https://bugs.python.org/issue31672
     idpattern = r'(?a:[_a-z][_a-z0-9]*)'
     braceidpattern = None
-    flags = _re.IGNORECASE
+    flags = None  # default: re.IGNORECASE
+
+    pattern = _TemplatePattern  # use a descriptor to compile the pattern
 
     def __init_subclass__(cls):
         super().__init_subclass__()
-        if 'pattern' in cls.__dict__:
-            pattern = cls.pattern
+        cls._compile_pattern()
+
+    @classmethod
+    def _compile_pattern(cls):
+        import re  # deferred import, for performance
+
+        cls_pattern = cls.__dict__.get('pattern')
+        if cls_pattern is not None and cls_pattern is not _TemplatePattern:
+            # Prefer a pattern defined on the class.
+            pattern = cls_pattern
         else:
```

Comment:
It can be simplified even more:
```suggestion
        pattern = cls.__dict__.get('pattern', _TemplatePattern)
        if pattern is _TemplatePattern:
```